android/app-newm/src/main/java/io/newm/Logout.kt

```diff
@@ -1,36 +1,48 @@
+@file:OptIn(DelicateCoroutinesApi::class)
+
 package io.newm
 
 import android.content.Context
 import co.touchlab.kermit.Logger
 import com.jakewharton.processphoenix.ProcessPhoenix
+import io.newm.shared.public.usecases.LoginUseCase
 import io.newm.shared.public.usecases.UserSessionUseCase
+import kotlinx.coroutines.DelicateCoroutinesApi
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
 import org.koin.core.component.KoinComponent
-import java.util.concurrent.Callable
 
 class Logout(
-    private val userSession: UserSessionUseCase,
+    private val loginUseCase: LoginUseCase,
+    private val userSessionUseCase: UserSessionUseCase,
     private val restartApp: RestartApp,
-) : KoinComponent, Callable<Int> {
+) : KoinComponent {
 
     private val logger = Logger.withTag("NewmAndroid-Logout")
 
-    override fun call(): Int {
-        return try {
-            userSession.logout()
-            restartApp.run()
-            logger.d("Logout successful")
-            0
-        } catch (e: Exception) {
-            logger.d("Logout Exception: $e")
-            1
-        }
+    fun signOutUser() = try {
+        loginUseCase.logout()
```

Comment:
probably should clean the song file cache when logging out too?

Reply:
The logout flows gets started on the client, so that is work we need to do before calling the `loginUseCase.logout()`

Android has not implemented the caching mechanism yet.

Reply:
But you download the song file don't you?  You're not streaming it?